pkg/api/composition_test.go

```diff
@@ -80,6 +89,57 @@ func TestValidateGroupBuildKey(t *testing.T) {
 	require.NotEqualValues(t, k3, k4)
 }
 
+func TestListBuilders(t *testing.T) {
+	c := &Composition{
+		Metadata: Metadata{},
+		Global: Global{
+			Plan:    "foo_plan",
+			Case:    "foo_case",
+			Builder: "docker:go",
+			Runner:  "local:docker",
+		},
+		Groups: []*Group{
+			{
+				ID:      "repeated",
+				Builder: "docker:generic",
+			},
+			{
+				ID: "another-id",
+			},
+		},
+	}
+
+	require.EqualValues(t, []string{"docker:generic", "docker:go"}, c.ListBuilders())
+}
+
+func TestBuildKeyWithoutGroupPanics(t *testing.T) {
```

Comment:
Aren't we testing that the opposite here? That calling `buildKey` panics if there is no builder?

Reply:
Good catch, renamed!